hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c

```diff
@@ -2448,6 +2448,67 @@ int launch_container_as_user(const char *user, const char *app_id,
     return exit_code;
 }
 
+static int parse_int (const char *str, int *val)
+{
+  long num;
+  char *endptr;
+
+  errno = 0;
+  num = strtol(str, &endptr, 0);
+
+  if (errno)
+    return -errno;
+
+  if (endptr == str || *endptr != '\0')
+    return -EINVAL;
+
+  if (num < INT_MIN || num > INT_MAX)
+    return -EINVAL;
+
+  *val = num;
+  return 0;
+}
+
+static void kill_session (pid_t session, int sig)
+{
+  DIR *d;
+  int ret;
+  struct dirent *de;
+  pid_t sid, pid = 0;
+
+  d = opendir("/proc");
+  if (!d) {
+    fprintf(LOGFILE, "Error opening /proc\n");
+    return;
+  }
+
+  for (;;) {
+    de = readdir(d);
+    if (!de)
+      break;
+
+    if (de->d_name[0] < '0' || de->d_name[0] > '9')
```

Comment:
could we add a safety net to also verify that de is a directory ? Is that relevant ?

Reply:
I can add that, but it's not really needed. The kernel doesn't export toplevel files with a numeric name in /proc.
Most main tools interacting with /proc (for example, procps) are not doing this check.